### DISCUSSION TITLE
How to render html without any hydration ?

### DISCUSSION THREAD

**Discussion by kristijorgji** [vercel/next.js#34538] [Help]
Hi let us assume we have a /blog page with 256 posts there.
I want to render the html via a reacct component for example Blog.tsx and return the html only without injecting in document object the hydration props inside __NEXT_DATA__

Reason for this is to cut in half the blog.html size as the props are not really needed.
In my example blog, I just render html via Blog.tsx of all posts and do not need any further js interaction.

How can we do such a thing with Next.JS ?

Before someone without enough information jumps and says this is not needed/stupid etc I would like to let you know that had done this in a custom server setup before. Where pre-created html for particular parts and only added the filter parts as react components to inject and worked great. Due to better code structure we migrated to next js and would like to do such thing as well. Normally creating custom server.ts file I assume would allow such thing as can use react server to renderHtmlToString manipulate somehow and remove thejs

I managed so far to remove the injection of props by overriding NextScript.getInlineScriptSource in _document.tsx like example
```
NextScript.getInlineScriptSource = (props) => {
    if (props.__NEXT_DATA__.props.pageProps && props.__NEXT_DATA__.props.pageProps[SSR_PAGE_CONFIG_PROP_NAME]) {
        const ssrConfig = (props.__NEXT_DATA__.props.pageProps as SinglePageSSRConfig).__s__;
        if (ssrConfig.dontHydrateProps) {
            for (const propName of ssrConfig.dontHydrateProps) {
                delete props.__NEXT_DATA__.props.pageProps[propName];
            }
        }
        delete props.__NEXT_DATA__.props.pageProps[SSR_PAGE_CONFIG_PROP_NAME];
    }

    return nextScript(props);
};
```

but then the component re-renders on page load (html returned is great as i want) and shows empty page

**Comment by Bahlaouane-Hamza** [comment]
Hi,
If you which to completely disable JS, you can use `unstable_runtimeJS `

But if you need to remove some data from `__NEXT_DATA__` and reduce your page size, you will need to patch Next.js for this, here's one for next@12.1.6
You can use https://www.npmjs.com/package/patch-package to auto install the patch.

[next+12.1.6.patch](https://github.com/vercel/next.js/files/11009343/next%2B12.1.6.patch)

For this part
```js
+    delete props1.pageProps.node
+    delete props1.pageProps.i18n
+    delete props1.pageProps.menus
```
I'm removing only the props i don't wanna keep in the HTML, If you are using NextAuth for example, you want to keep `session` coming from SSR, and this may vary depending on your use case.

And finally on the _hydrate part, i just restore the props i removed.

Hope this helps.
